src/celldega/nbhd/__init__.py

```diff
@@ -300,4 +305,208 @@ def create_hextile(radius, path_landscape_files=None, img_height=100, img_width=
+        in each hexagonal tile and saves the intermediate transcript assignment result as a Parquet file.
+
+    """
+
+    from ..pre.boundary_tile import batch_transform_geometries
+    from ..pre.__init__ import _to_geometry
+
+    print("Reading transcripts file...")
+
+    ## only Xenium and Custom Tech supported for now
+
+    if gdf_transcripts is None:
+
+        if path_data is None:
+            raise ValueError("Either gdf_transcripts or path_data must be provided.")
+
+        trx = read_parquet_error_check(os.path.join(path_data, "transcripts.parquet"))
+
+        if os.path.isfile(os.path.join(path_data, "experiment.xenium")):
+            technology = "Xenium"
+
+            x = "x_location"
+            y = "y_location"
+            cell_id_col = "cell_id"
+            segmentation_approach = "default"
+
+            gdf_trx = gpd.GeoDataFrame(trx, geometry=gpd.points_from_xy(trx[x], trx[y]))
+
+        elif os.path.isfile(os.path.join(path_data, "segmentation_parameters.json")):
+
+            with open(os.path.join(path_data, "segmentation_parameters.json"), "r") as parameters_file:
+                parameters = json.load(parameters_file)
+                technology = parameters["technology"]
+
+            x = "x"
+            y = "y"
+            cell_id_col = "cell_index"
+            segmentation_approach = parameters["segmentation_approach"]
+
+            if type(trx['geometry'].iloc[0]) == str:
+                trx['geometry'] = trx['geometry'].apply(wkt.loads)
+
+            gdf_trx = gpd.GeoDataFrame(trx, geometry="geometry")
+            gdf_trx.set_geometry("geometry", inplace=True)
+
+        else:
+            raise ValueError("Unsupported or missing transcriptomics technology files.")
+
+    else:
+        gdf_trx = gdf_transcripts.copy()
+        gdf_trx.set_geometry("geometry", inplace=True)
+
+        technology = "PYTEST"
+        cell_id_col = "cell_index"
+        x = "x"
+        y = "y"
+
+    print("Transcripts file read.")
+    print("Assignment of transcripts started...")
+
+    gdf_hextile.set_geometry("geometry", inplace=True)
+
+    hextile_assigned_trx = gpd.sjoin(
+        gdf_trx, gdf_hextile, how="left", predicate="within"
+    )
+
+    hextile_assigned_trx.rename(columns={"index_right": "polygon_index", "geometry_image_space_left": "geometry_image_space"}, inplace=True)
+    hextile_assigned_trx.drop(["geometry_image_space"], axis=1, inplace=True)
+
+    hextile_assigned_trx["polygon_index"] = (
+        hextile_assigned_trx["polygon_index"].astype(str) + "_polygon"
+    )
+
+    gdf_hextile_assigned_trx = gpd.GeoDataFrame(
+        hextile_assigned_trx, geometry="geometry"
+    )
+
+    if technology != "PYTEST":
+
+        gdf_hextile_assigned_trx.to_parquet(
+        os.path.join(
+            path_landscape_files,
+            f"hextile_assigned_trx_{technology}_{segmentation_approach}.parquet")
+        )
+
+    print("Assignment of transcripts done and saved.")
```

Comment:
Can we move the transcript assignment to hexagons to another function so that we can reuse it when we calculate the nbg matrix (neighborhood-by-gene matrix)?